.buildbot.sh

```diff
@@ -1,11 +1,15 @@
 #!/bin/bash
 set -e
 
-CHERIBUILD=~/build
+set ${CHERIBUILD:="/home/buildbot/build"}
+set ${CHERI_DIR:="/home/buildbot/cheri/output"}
+set ${SSHPORT:=10021}
+set ${SSHHOST:=localhost}
+set ${PYTHONPATH:=${CHERIBUILD}/test-scripts}
 
-export SSHPORT=10021
-export SSHHOST=localhost
-export PYTHONPATH="$CHERIBUILD"/test-scripts
+export SSHPORT
```

Comment:
I've never seen this `set` > `export` pattern before. Do the `set` commands only update the values if there are no environment variables with that name yet, and then the `export` commands perform an update?

Reply:
Yes, if the environment variables have been already set, those are used. Otherwise, they will default to the values set here. The `export` is to make sure that all sub-shells will get that variable.

Reply:
Isn't this just a two step version of `export SSHPORT=10021` and so on? [I might be wrong! The Unix shell has some weird corner cases...]

Reply:
Yes - mostly. But I thought @0152la  wanted to ensure that if these variables were being set outside this script, those values needed to be used rather than the defaults that are defined here.

Reply:
Looking at the `webkit` repo, and with a small test, it does seem the changes propagate in subshells without the need for export.

```
echo $E_VAR
E_VAR=${E_VAR-mimi}
/bin/bash -c "echo $E_VAR"
```


[1] https://github.com/capablevms/webkit/pull/1/files#diff-0693c60a7dfecde2f66e3572d6a12d454a8382abf4ab33653e4814929dfb6b94R9

Reply:
We now think the `set` makes things local which is why the `export` is then needed.